gemini/sample-apps/fashion-trends-prediction/app/pages/2_     ● Fashion Bot.py

```diff
@@ -0,0 +1,231 @@
+
+add_logo(config['Images']['logo'])
+
+
+PROJECT_ID = config["PROJECT_ID"]  # @param {type:"string"}
+LOCATION = config["LOCATION"]  # @param {type:"string"}
+data_path = config["Data"]['current_data']
+
+generation_config = config['parameters']['fashion_bot']
+safety_settings = {
+    generative_models.HarmCategory.HARM_CATEGORY_HATE_SPEECH: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+    generative_models.HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+    generative_models.HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+    generative_models.HarmCategory.HARM_CATEGORY_HARASSMENT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+}
+
+
+st.image(image=stImg(config["Images"]["chat"]), width=200)
+
+if 'JSONdata' not in st.session_state:
+    vertexai.init(project=PROJECT_ID, location=LOCATION)
+
+    with open(data_path, 'r') as f:
+        st.session_state['JSONdata'] = json.load(f)
+
+
+country = st.selectbox(
+    'Select Country',
+    st.session_state['JSONdata']['finaldata'].keys()
+)
+
+category = st.selectbox(
+    'Select category of outfit?',
+    list(map(lambda x: x.capitalize(),
+         st.session_state['JSONdata']['top_categories'][country]))
+)
+
+
+prompt = st.chat_input("Ask Something")
+
+if prompt:
+    st.session_state['prompt'] = prompt
+    st.rerun()
+
+
+category = category.lower()
+
+cat_key = 'chat-context' + country + category
+
+if cat_key in st.session_state:
+    context = st.session_state[cat_key]
+else:
+    context = f"""You are a fashion expert in {category}. Answer fashion related queries solely based on the data provided.
+
+    Important points -
+    - The data provided represents what popular influencers are wearing nowadays
+    - If user asks about something that is not present in the data, answer that it is not trending in fashion
+    - Popularity of an item is proportional to its counts in the data
+    - If item is not in data provided, it is not fashionable.
+
+    {category.capitalize()} data - """
+
+    words = 0
+    if category in st.session_state['JSONdata']['finaldata'][country]:
+        for outfit in st.session_state['JSONdata']['finaldata'][country][category]:
+            if isinstance(outfit, str):
+                context += outfit + ', '
+                words += 1
+            if words > 3000:
+                break
+
+    st.session_state[cat_key] = context
+
+bot_key = 'bot' + country + category
+
+
+if bot_key in st.session_state:
+    bot = st.session_state[bot_key]
+else:
+    model = GenerativeModel("gemini-1.0-pro-002")
+    bot = model.start_chat()
+
+    st.session_state[bot_key] = bot
+
+
+hist_key = 'chatHistory' + country + category
+img_gen_one_shot = ""
+
+
+if hist_key not in st.session_state:
+
+    with st.spinner("Loading Chatbot..."):
+        bot.send_message(st.session_state[cat_key])
```

Comment:
A lot of this could be sent in System Prompts. I don't think you need to send multiple messages for this.

https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/gemini#:~:text=seconds%22%3A%2070%20%7D.-,systemInstruction,-(Preview)

Reply:
Also, add these to your prompts file or a new frontend_prompts.py file

Reply:
Added these instructions in the system instructions. The context (which is sent to system instruction) is moved to file genai_prompts.py